### DISCUSSION TITLE
[Linux] Purpose of /etc/boinc-client in Fedora

### DISCUSSION THREAD

**Discussion by gj82854** [BOINC/boinc#5334] [Q&A] [answered]
When testing the Linux install process for 7.24.x I noticed that there is an /etc/boinc-client directory which I didn't think existed in the Fedora world. I went back to some previous releases and there is an /etc/boinc-client directory in those releases too. Looking into the contents, there was only one file, config.properties (boinc.conf in 7.24.x). The config.properties just contained the DATA_DIR=/var/lib/boinc and the boinc.conf contained the init.d start variables but they were all commented out. What is the purpose for this directory in the Fedora world and is it destined to stay or be removed? Maybe backward compatibility? Neither seem to provide a real purpose as the same data is contained in the systemd services file for the most part. 

UPDATE: It seems to be needed for the Manager. Without it, the manager can't find the data_dir and issues a message that it can't find gui.rpc.auth.cfg.  Based on this information i'm going to open a new issue for the 7.24.x install. Manager can't find the DATA_DIR with the current boinc.conf (which currently contains nothing but comments)

**Comment by AenBleidd** [comment, marked as answer]
We now provide vanilla packages for various Linux distros (including Fedora). Feel free to use them instead: https://boinc.berkeley.edu/linux_install.php

**Comment by gj82854** [comment]
The packages do not provide the necessary information in /etc/boinc-client to allow boinccmd to work.
Install of Fedora 40 packages do not work. Instead of installing 8.0.3 the installation process say that 7.xx.x is already installed and there is nothing to do.

Sent with [Proton Mail](https://proton.me/) secure email.

**Comment by AenBleidd** [comment]
@gj82854, have you followed the instructions from the link above?

**Comment by gj82854** [comment]
yes sir, I did. I had the instructions opened while I entered the commands into a terminal

Sent with [Proton Mail](https://proton.me/) secure email.

**Comment by gj82854** [comment]
I was wondering if the PRIORITY=100 parameter in the repository might be the problem. Doesn't that mean it's the lowest priority? However, I commented out the parameter and it still gave the same results

Sent with [Proton Mail](https://proton.me/) secure email.

**Comment by gj82854** [comment]
Followed the instructions: (NOTE: this is the alpha repository)

***@***.***:/etc/yum.repos.d$ sudo dnf install dnf-plugins-core
Last metadata expiration check: 2:30:50 ago on Fri 26 Jul 2024 05:11:11 AM CDT.
Package dnf-plugins-core-4.8.0-1.fc40.noarch is already installed.
Dependencies resolved.
Nothing to do.
Complete!
***@***.***:/etc/yum.repos.d$ sudo dnf config-manager --add-repo https://boinc.berkeley.edu/dl/linux/alpha/fc40
Adding repo from: https://boinc.berkeley.edu/dl/linux/alpha/fc40
***@***.***:/etc/yum.repos.d$ sudo dnf config-manager --set-enabled boinc.berkeley.edu/dl/linux/alpha/fc40
Error: No matching repo to modify: boinc.berkeley.edu/dl/linux/alpha/fc40.
***@***.***:/etc/yum.repos.d$ sudo dnf config-manager --set-enabled boinc.berkeley.edu_dl_linux_alpha_fc40
***@***.***:/etc/yum.repos.d$ sudo rpm --import https://boinc.berkeley.edu/dl/linux/alpha/fc40/boinc.gpg
***@***.***:/etc/yum.repos.d$ sudo yum install boinc-client
created by dnf config-manager from https://boin 3.2 kB/s | 1.8 kB 00:00
Package boinc-client-7.20.2-9.fc40.x86_64 is already installed.
Dependencies resolved.
Nothing to do.
***@***.***:/etc/yum.repos.d$

Sent with [Proton Mail](https://proton.me/) secure email.

**Reply by AenBleidd** [reply]
Looks like yum doesn't allow to update a package from the different repo without some manipulations.
You can try this method: https://serverfault.com/questions/476776/yum-the-same-packages-in-two-repositories-force-to-install-the-package-from-a

**Comment by gj82854** [comment]
I could do that but that seems like more of a work-around than a solution. There will always be BOINC in the distribution's repos unless you prevent them from picking it up. This also doesn't allow automatic updates during maintenance cycles.

BTW: Did you yum is deprecated? Replaced by DNF

Sent with [Proton Mail](https://proton.me/) secure email.
